weather_mv/loader_pipeline/ee.py

```diff
@@ -497,6 +499,20 @@ def __init__(self,
         self.ee_asset = ee_asset
         self.ee_asset_type = ee_asset_type
 
+    def ee_tasks_remaining(self) -> int:
+        """Returns the remaining number of tasks in the tassk queue of earth engine."""
+        return len([task for task in ee.data.getTaskList()
+                    if task['state'] in ["UNSUBMITTED", "READY", "RUNNING"]])
+
+    def wait_for_task_queue(self) -> None:
+        """Waits until the task queue has space.
+
+        Ingestion of table in the earth engine creates a task and every project has a limited task queue size. This
+        function checks the task queue size and waits until the task queue has some space.
+        """
+        while self.ee_tasks_remaining() >= self._num_shards:
+            time.sleep(TASK_QUEUE_WAIT_TIME)
```

Comment:
An aside, this may be unfavorable to the autoscaling algorithm. I'm happy to merge this now, but if we experience performance problems, we may need to use a more beam-friendly solution. 

https://beam.apache.org/documentation/programming-guide/#processing-time-timers